include/grpcpp/impl/codegen/server_callback.h

```diff
@@ -435,23 +429,25 @@ class ServerUnaryReactor : public internal::ServerReactor {
 };
 
 template <typename Request, typename Response, typename F>
-ServerUnaryReactor<Request, Response>* ServeRpc(F&& func) {
+ServerUnaryReactor<Request, Response>* ServeRpc(ServerContext* context,
+                                                F&& func) {
   // TODO(vjpai): Specialize this to prevent counting OnCancel conditions
   class SimpleUnaryReactor final
       : public ServerUnaryReactor<Request, Response> {
    public:
     explicit SimpleUnaryReactor(F&& func) : on_started_func_(std::move(func)) {}
 
    private:
-    void OnStarted(ServerContext* context, const Request* req,
-                   Response* resp) override {
-      on_started_func_(context, req, resp, this);
+    void OnStarted(const Request* req, Response* resp) override {
+      on_started_func_(req, resp, this);
```

Comment:
The user still has to call Finish to indicate done-ness. We could have a "super-simple-reactor" where the user returns a status and the library calls Finish; it could even be an overload...

Reply:
I guess it'll be a different function name rather than an overload since I think the right type_trait to separate these function types is is_invocable which is not valid until C++17 (which we have not yet committed to use)

Reply:
Never mind, enable_if + decltype to the rescue. We can do this as an overload.